kratos/modeler/create_entities_from_geometries_modeler.cpp

```diff
@@ -0,0 +1,110 @@
+
+
+namespace Kratos
+{
+
+    template <>
+    void CreateEntitiesFromGeometriesModeler::RemoveModelPartEntities<Element>(ModelPart &rModelPart)
+    {
+        const SizeType n_elements = rModelPart.NumberOfElements();
+        KRATOS_WARNING_IF("CreateEntitiesFromGeometriesModeler", n_elements != 0)
+            << "There are " << n_elements << " elements in '" << rModelPart.FullName() << "' model part. These are to be removed." << std::endl;
+        VariableUtils().SetFlag(TO_ERASE, true, rModelPart.Elements());
+        rModelPart.RemoveElementsFromAllLevels(TO_ERASE);
+    }
+
+    template <>
+    void CreateEntitiesFromGeometriesModeler::RemoveModelPartEntities<Condition>(ModelPart &rModelPart)
+    {
+        const SizeType n_conditions = rModelPart.NumberOfConditions();
+        KRATOS_WARNING_IF("CreateEntitiesFromGeometriesModeler", n_conditions != 0)
+            << "There are " << n_conditions << " conditions in '" << rModelPart.FullName() << "' model part. These are to be removed." << std::endl;
+        VariableUtils().SetFlag(TO_ERASE, true, rModelPart.Conditions());
+        rModelPart.RemoveConditionsFromAllLevels(TO_ERASE);
+    }
+
+    template <>
+    void CreateEntitiesFromGeometriesModeler::LoopEntitiesList<Element>(Parameters EntitiesList)
+    {
+        for (auto& r_data : EntitiesList) {
+            // Get current substitution settings
+            const std::string entity_name = r_data["element_name"].GetString();
+            const std::string model_part_name = r_data["model_part_name"].GetString();
+            auto &r_model_part = mpModel->GetModelPart(model_part_name);
+
+            // Wipe current model part entities
+            RemoveModelPartEntities<Element>(r_model_part);
+
+            // Create submodelpart elements from geometries
+            CreateEntitiesFromGeometries<Element, ModelPart::ElementsContainerType>(entity_name, r_model_part);
+        }
+    }
+
+    template <>
+    void CreateEntitiesFromGeometriesModeler::LoopEntitiesList<Condition>(Parameters EntitiesList)
+    {
+        for (auto& r_data : EntitiesList) {
+            // Get current substitution settings
+            const std::string entity_name = r_data["condition_name"].GetString();
+            const std::string model_part_name = r_data["model_part_name"].GetString();
+            auto &r_model_part = mpModel->GetModelPart(model_part_name);
+
+            // Wipe current model part entities
+            RemoveModelPartEntities<Condition>(r_model_part);
+
+            // Create submodelpart conditions from geometries
+            CreateEntitiesFromGeometries<Condition, ModelPart::ConditionsContainerType>(entity_name, r_model_part);
+        }
+    }
+
+    void CreateEntitiesFromGeometriesModeler::SetupModelPart()
+    {
+        // Check parallelism
+        //TODO: Make this MPI compatible by renumbering the local ids
+        KRATOS_ERROR_IF(ParallelEnvironment::GetDefaultDataCommunicator().IsDistributed())
```

Comment:
please check the ModelPart that you pass in the input, not the global parallelism, otherwise it will be incompatible with SubCommunicators

Reply:
I did it like this to avoid checking it for each submodelpart. In any case, checking it for each one shouldn't be a problem so OK.